src/js/hooks/useTableDataV2.js

```diff
@@ -69,9 +71,16 @@ const useTableDataV2 = ({
 
   // fetching data after changing page size, filters, page number and sorting
   useEffect(() => {
-    if (shouldFetch) {
-      fetchData();
+    if (!shouldFetch) {
+      return;
     }
+    // Prevents automatic data fetching when resetInitialFetch is true,
+    // allowing fetch only on submit button click
+    if (resetInitialFetch && setResetInitialFetch) {
```

Comment:
why do we check `setResetInitialFetch`? Isn't `resetInitialFetch` enough?

Reply:
If resetInitialFetch blocks from the initial fetch, then the naming isn't appropriate